sdk/core/Azure.Core/src/Shared/MessagingClientDiagnostics.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Azure.Core.Pipeline;
+
+#nullable enable
+
+namespace Azure.Core.Shared
+{
+    /// <summary>
+    /// Client Diagnostics support for messaging clients. Currently, this is only used for AMQP clients.
+    /// HTTP libraries should use the ClientDiagnostics type instead.
+    /// </summary>
+    internal class MessagingClientDiagnostics
+    {
+        private readonly string _fullyQualifiedNamespace;
+        private readonly string _entityPath;
+        private readonly string _messagingSystem;
+        private readonly DiagnosticScopeFactory _scopeFactory;
+
+        #region OTel-specific messaging attributes
+        public const string MessagingSystem = "messaging.system";
+        public const string DestinationName = "messaging.destination.name";
+        public const string SourceName = "messaging.source.name";
+        public const string MessagingOperation = "messaging.operation";
+        public const string NetPeerName = "net.peer.name";
+        public const string BatchCount = "messaging.batch.message_count";
+        #endregion
+
+        #region legacy compat attributes
+        public const string MessageBusDestination = "message_bus.destination";
+        public const string PeerAddress = "peer.address";
+        public const string Component = "component";
+        #endregion
+
+        public const string DiagnosticIdAttribute = "Diagnostic-Id";
+
+        public MessagingClientDiagnostics(string clientNamespace, string? resourceProviderNamespace, string messagingSystem, string fullyQualifiedNamespace, string entityPath)
```

Comment:
Do any of these args need to be validated for null?

Reply:
Since this is internal it seems like it is okay to skip validation.